File: app/src/main/java/com/gooner10/popularmoviesapp/Activity/moviemain/MovieContract.java
package com.gooner10.popularmoviesapp.Activity.moviemain;

import com.gooner10.popularmoviesapp.Activity.data.MovieData;

import java.util.ArrayList;
import java.util.List;

/**
 * This specifies the contract between the Movie view and the Movie presenter.
 */
public interface MovieContract {
    interface View {
        void displayMovieList(List<MovieData> movieDataList);
    }

    interface UserActionsListener {
        void loadData();
    }
}

File: feature/moviehome/src/main/java/com/gooner10/moviehome/MovieContract.kt
package com.gooner10.moviehome

import com.gooner10.data.model.MovieItem


/**
 * This specifies the contract between the Movie view and the Movie presenter.
 */
interface MovieContract {
    interface View {
        fun displayMovieList(movieDataList: List<MovieItem>)
    }
}
